### ISSUE TITLE
ORA-28009: connection as SYS should be as SYSDBA or SYSOPER error for OCI_PoolCreate

### ISSUE THREAD

**Issue by jssprasanna** [vrogier/ocilib#222]
We can try below-mentioned code to reproduce the issue.

#include "ocilib.h"
#define MAX_THREADS 50
#define MAX_CONN    10
#define SIZE_STR   260
void err_handler(OCI_Error *err)
{
    printf("%s\n", OCI_ErrorGetString(err));
}
void pool_worker(OCI_Thread *thread, void *data)
{
    OCI_Connection *cn = OCI_PoolGetConnection(data, NULL);
    char str[SIZE_STR + 1] = "";
    OCI_Immediate(cn, "select to_char(sysdate, 'YYYYMMDD HH24:MI:SS') from dual", OCI_ARG_TEXT, str);
    printf("%s\n", str);
    OCI_ConnectionFree(cn);
}
int main(void)
{
    OCI_Thread *th[MAX_THREADS];
    OCI_ConnPool *pool;
    int i;
    if (!OCI_Initialize(err_handler, NULL, OCI_ENV_DEFAULT | OCI_ENV_THREADED))
    {
        return EXIT_FAILURE;
    }
    /* create pool */
    pool = OCI_PoolCreate("db", "sys", "password", OCI_POOL_SESSION, OCI_SESSION_SYSDBA, 0, MAX_CONN, 1);
    /* create threads */
    for (i = 0; i < MAX_THREADS; i++)
    {
        th[i] = OCI_ThreadCreate();
        OCI_ThreadRun(th[i], pool_worker, pool);
    }
    /* wait for threads and cleanup */
    for (i = 0; i < MAX_THREADS; i++)
    {
        OCI_ThreadJoin(th[i]);
        OCI_ThreadFree(th[i]);
    }
    OCI_PoolFree(pool);
    OCI_Cleanup();
    return EXIT_SUCCESS;
}

**Comment by vrogier** [comment]
Hi,

I will check if the issue is in OCILIB code or if it is an Oracle limitation.
Using sys account for connection pools seems quite unusual as sys account should only be used for specific DB administration operations.

Regards,

Vincent

**Comment by vrogier** [comment]
Hi,

It appears that when session pooling was introduced in oracle 9i, connecting as sysdba was not possible/documented in OCI documentation.
In the meanwhile oracle has added the possibility to do (new flag OCI_SESSGET_SYSDBA for OCI call OCISessionGet() ).
I did not implemented it as it was not available at the time I added support for session pools years ago. 
I regularly check for OCI changes and I missed that one. 
I will add support for it and commit tomorrow in a new development branch for v4.6.4 release.

Best regards,

Vincent
